sdk/monitor/monitor-query/src/models/publicMetricsModels.ts

```diff
@@ -0,0 +1,131 @@
+export interface Metric {
+  /** the metric Id. */
+  id: string;
+  /** the resource type of the metric resource. */
+  type: string;
+  /** the name and the display name of the metric, i.e. it is localizable string. */
+  name: LocalizableString;
+  /** Detailed description of this metric. */
+  displayDescription: string;
+  /** 'Success' or the error details on query failures for this metric. */
+  errorCode?: string;
+  /** the unit of the metric. */
+  unit: MetricUnit;
+  /** the time series returned when a data query is performed. */
+  timeseries: TimeSeriesElement[];
+}
+
+/** A time series result type. The discriminator value is always TimeSeries in this case. */
+export interface TimeSeriesElement {
+  // track 2 version of
+
+  /** the metadata values returned if $filter was specified in the call. */
+  metadataValues?: MetadataValue[];
+  /** An array of data points representing the metric values.  This is only returned if a result type of data is specified. */
+  data?: MetricValue[];
+}
+
+/**
+ * Metrics, including additional information like cost, the resourceRegion, etc...
+ */
+export interface QueryMetricsResponse {
+  // track 2 version of `MetricsListResponse`
+
+  /** The integer value representing the cost of the query, for data case. */
+  cost?: number;
+  /** The timespan for which the data was retrieved. Its value consists of two datetimes concatenated, separated by '/'.  This may be adjusted in the future and returned back from what was originally requested. */
+  timespan: string;
+  /** The interval (window size) for which the metric data was returned in.  This may be adjusted in the future and returned back from what was originally requested.  This is not present if a metadata request was made. */
+  interval?: string;
+  /** The namespace of the metrics been queried */
+  namespace?: string;
+  /** The region of the resource been queried for metrics. */
+  resourceRegion?: string;
+  /** the value of the collection. */
+  metrics: Metric[];
+}
+
+/**
+ * Options used when getting metric definitions.
+ */
+export interface GetMetricDefinitionsOptions extends OperationOptions {
+  // track 2 version of `MetricDefinitionsListOptionalParams`
+
+  /** Metric namespace to query metric definitions for. */
+  metricNamespace?: string;
+}
+
+/**
+ * Metric definitions.
+ */
+export interface GetMetricDefinitionsResponse {
+  /** the values for the metric definitions. */
+  definitions: MetricDefinition[];
+}
+
+/**
+ * Options used when getting metric namespaces.
+ */
+export interface GetMetricNamespacesOptions {
+  // track 2 copy of `MetricNamespacesListOptionalParams`
+
+  /** The ISO 8601 conform Date start time from which to query for metric namespaces. */
+  startTime?: string;
+}
+
+/**
+ * Metric namespaces.
+ */
+export interface GetMetricNamespacesResponse {
+  // track 2 version of MetricNamespacesListResponse
+
+  /** The metric namespaces. */
+  namespaces: MetricNamespace[] | undefined;
```

Comment:
Is this field really always present, or should it be an optional field?

Reply:
Fixed - it should always be there.